src/Tap.sol

```diff
@@ -75,21 +87,27 @@ contract Tap {
         require(now == rho, "Tap/rho-not-updated");
         if (what == "flight") flight = data;
         else revert("Tap/file-unrecognized-param");
+
+        emit File(what, data);
     }
     function file(bytes32 what, uint256 data) external auth {
         require(now == rho, "Tap/rho-not-updated");
         if (what == "rate") rate = data;
         else revert("Tap/file-unrecognized-param");
+
+        emit File(what, data);
     }
 
+    // --- External ---
     function pump() external stoppable {
-        require(now >= rho, "Tap/invalid-now");
-        uint256 wad = mul(now - rho, rate);
-        if (wad > 0) {
-            vat.suck(address(vow), address(this), mul(wad, RAY));
-            daiJoin.exit(address(this), wad);
-            keg.pour(flight, wad);
-        }
+        require(now > rho, "Tap/invalid-now");
+        uint256 wad = mul(now - rho, rate);  // TODO How is that working? We clearly need to increase the test coverage
         rho = now;
+
+        require(wad > 0, "Tap/invalid-wad");
```

Comment:
This is unnecessary as if now - rho > 0 then wad must also be > 0 unless rate is 0 for some reason, but it will just end up being a no-op for the rest of the calls.

Reply:
I would exclude it. I don't like to add unnecessary requires for the common path.

Reply:
removed it by the latest commit.